qiskit_machine_learning/algorithms/distribution_learners/qgan/quantum_generator.py

```diff
@@ -417,10 +421,22 @@ def gradient_function(current_point):
                 quantum_instance, params=current_point, shots=self._shots
             )
             prediction_generated = discriminator.get_label(generated_data, detach=True)
-            op = ~CircuitStateFn(primitive=self.generator_circuit)
+            op = CircuitStateFn(primitive=self.generator_circuit)
             grad_object = gradient_object.convert(operator=op, params=free_params)
             value_dict = {free_params[i]: current_point[i] for i in range(len(free_params))}
-            analytical_gradients = np.array(grad_object.assign_parameters(value_dict).eval())
+            if quantum_instance.backend is not None:
+                grad_object = (
+                    CircuitSampler(quantum_instance.backend).convert(grad_object, value_dict).eval()
+                )
+                if quantum_instance.is_statevector:
+                    analytical_gradients = np.array(grad_object)
```

Comment:
Preferable way is to use `np.asarray()`. This function does not copy data if possible and a call to `np.array()` like here copies data.
```suggestion
                    analytical_gradients = np.asarray(grad_object)
```